frontend/controller/app/identity.js

```diff
@@ -55,96 +56,106 @@ sbp('okTurtles.events/on', LOGIN, async ({ identityContractID, encryptionParams,
+      }
 
-    // NOTE: users could notice that they leave the group by someone
-    // else when they log in
-    const currentState = sbp('state/vuex/state')
-    if (!currentState.currentGroupId) {
-      const gId = Object.keys(currentState.contracts)
-        .find(cID => currentState[identityContractID].groups[cID] && !currentState[identityContractID].groups[cID].hasLeft)
+      // NOTE: users could notice that they leave the group by someone
+      // else when they log in
+      const currentState = sbp('state/vuex/state')
+      if (!currentState.currentGroupId) {
+        const gId = Object.keys(currentState.contracts)
+          .find(cID => currentState[identityContractID].groups[cID] && !currentState[identityContractID].groups[cID].hasLeft)
 
-      if (gId) {
-        sbp('gi.app/group/switch', gId)
+        if (gId) {
+          sbp('gi.app/group/switch', gId)
+        }
       }
+
+      // Whenever there's an active session, the encrypted save state should be
+      // removed, as it is only used for recovering the state when logging in
+      sbp('gi.db/settings/deleteEncrypted', identityContractID).catch(e => {
+        console.error('Error deleting encrypted settings after login')
+      })
+
+      /* Commented out as persistentActions are not being used
+      // TODO: [SW] It may make more sense to load persistent actions in
+      // actions in the SW instead of on each tab
+      const databaseKey = `chelonia/persistentActions/${identityContractID}`
+      sbp('chelonia.persistentActions/configure', { databaseKey })
+      await sbp('chelonia.persistentActions/load')
+      */
+
+      sbp('okTurtles.events/emit', LOGIN_COMPLETE, { identityContractID })
+    } catch (e) {
+      sbp('okTurtles.events/emit', LOGIN_ERROR, { identityContractID, error: e })
     }
+  })
+})
 
-    // Whenever there's an active session, the encrypted save state should be
-    // removed, as it is only used for recovering the state when logging in
-    sbp('gi.db/settings/deleteEncrypted', identityContractID).catch(e => {
-      console.error('Error deleting encrypted settings after login')
-    })
+sbp('okTurtles.events/on', NEW_UNREAD_MESSAGES, (currentChatRoomUnreadMessages) => {
+  sbp('state/vuex/commit', 'setUnreadMessages', currentChatRoomUnreadMessages)
+})
 
-    /* Commented out as persistentActions are not being used
-    // TODO: [SW] It may make more sense to load persistent actions in
-    // actions in the SW instead of on each tab
-    const databaseKey = `chelonia/persistentActions/${identityContractID}`
-    sbp('chelonia.persistentActions/configure', { databaseKey })
-    await sbp('chelonia.persistentActions/load')
-    */
-
-    sbp('okTurtles.events/emit', LOGIN_COMPLETE, { identityContractID })
-  } catch (e) {
-    sbp('okTurtles.events/emit', LOGIN_ERROR, { identityContractID, error: e })
-  }
+sbp('okTurtles.events/on', NEW_PREFERENCES, (preferences) => {
+  sbp('state/vuex/commit', 'setPreferences', preferences)
 })
```

Comment:
RFC: above these events here a comment that says where these events are coming from, e.g.:

```js
// handle incoming identity-related events that are sent from the service worker
```